lsp/source/lib/typescript-service.mts

```diff
@@ -542,11 +543,15 @@ function TSService(projectURL = "./") {
   }
 
   function transpileCivet(path: string, source: string) {
-    return Civet.compile(source, {
-      ...civetConfig,
-      filename: path,
-      sourceMap: true,
-    })
+    const errors: Error[] = [],
+      result = Civet.compile(source, {
+        ...civetConfig,
+        filename: path,
+        sourceMap: true,
+        errors,
+      })
+
+    return Object.assign(result, { errors })
```

Comment:
Seeing this interface makes me wonder whether `Civet.compile` should directly return `errors` along with code and sourceMap, instead of (or in addition to?) modifying options. Is the issue that `compile` doesn't return an object when not requesting sourceMap?

Reply:
Yes, I think in the long run the errors can be returned alongside the sourceMap. When not requesting a sourceMap we should still throw because it only returns the compiled code as a string.